File: clients/apple/SwiftLockbookCore/Sources/SwiftLockbookCore/Models/Drawing.swift
public struct Drawing: Codable {
    public let scale: Float
    public let translationX: Float
    public let translationY: Float
    public let strokes: [Stroke]

    public init() {
        scale = 1.0
        translationX = 1.0
        translationY = 1.0
        strokes = []
    }

    public init(scale: Float, translationX: Float, translationY: Float, strokes: [Stroke]) {
        self.scale = scale
        self.translationX = translationX
        self.translationY = translationY
        self.strokes = strokes
    }
}

public struct Stroke: Codable {
    public let pointsX: [Float]
    public let pointsY: [Float]
    public let pointsGirth: [Float]
    public let color: ColorAlias
    public let alpha: Float

    public init(pointsX: [Float], pointsY: [Float], pointsGirth: [Float], color: ColorAlias, alpha: Float) {
        self.pointsX = pointsX
        self.pointsY = pointsY
        self.pointsGirth = pointsGirth
        self.color = color
        self.alpha = alpha
    }
}

public enum ColorAlias: String, Codable {
    case Black
    case Blue
    case Cyan
    case Green
    case Magenta
    case Red
    case White
    case Yellow
}

File: libs/lb/lb-rs/src/shared/drawing.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

#[derive(Clone, Serialize, Deserialize, Debug)]
pub struct Drawing {
    pub scale: f32,
    pub translation_x: f32,
    pub translation_y: f32,
    pub strokes: Vec<Stroke>,
    pub theme: Option<HashMap<ColorAlias, ColorRGB>>,
}

impl Default for Drawing {
    fn default() -> Self {
        Drawing { scale: 1.0, translation_x: 0.0, translation_y: 0.0, strokes: vec![], theme: None }
    }
}

#[derive(Clone, Serialize, Deserialize, Debug)]
pub struct Stroke {
    pub points_x: Vec<f32>,
    pub points_y: Vec<f32>,
    pub points_girth: Vec<f32>,
    pub color: ColorAlias,
    pub alpha: f32,
}

impl Stroke {
    pub fn new(color: ColorAlias) -> Self {
        Self {
            points_x: Vec::new(),
            points_y: Vec::new(),
            points_girth: Vec::new(),
            color,
            alpha: 1.0,
        }
    }

    pub fn is_empty(&self) -> bool {
        self.points_x.is_empty() && self.points_y.is_empty() && self.points_girth.is_empty()
    }
}

#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone, Copy)]
pub enum ColorAlias {
    Black,
    Red,
    Green,
    Yellow,
    Blue,
    Magenta,
    Cyan,
    White,
}

#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Hash)]
pub struct ColorRGB {
    pub r: u8,
    pub g: u8,
    pub b: u8,
}
